### PR TITLE
Modernize types casting

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | /
| Type?         | improvement
| Category?     | CO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | /
| How to test?  | /

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/11708)

### CODE REVIEW

File: controllers/admin/AdminSearchConfController.php [hunk 1]
```diff
@@ -371,8 +371,8 @@ public function renderForm()
 
     public function processSave()
     {
-        $search = strval(Tools::getValue('search'));
-        $string = strval(Tools::getValue('alias'));
+        $search = (string) (Tools::getValue('search'));
+        $string = (string) (Tools::getValue('alias'));
```
Comment (Quetzacoalt91) [inline controllers/admin/AdminSearchConfController.php]: This is also true is another places of this PR, some parentheses can be safely removed.
```suggestion
        $string = (string) Tools::getValue('alias');
```

File: controllers/admin/AdminSearchConfController.php [hunk 2]
```diff
@@ -371,8 +371,8 @@ public function renderForm()
 
     public function processSave()
     {
-        $search = strval(Tools::getValue('search'));
-        $string = strval(Tools::getValue('alias'));
+        $search = (string) (Tools::getValue('search'));
```
Comment (Quetzacoalt91) [inline controllers/admin/AdminSearchConfController.php]:
```suggestion
        $search = (string) Tools::getValue('search');
```

File: controllers/front/GetFileController.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function init()
             if (!Validate::isSha1($filename)) {
                 die(Tools::displayError());
             }
-            $file = _PS_DOWNLOAD_DIR_ . strval(preg_replace('/\.{2,}/', '.', $filename));
+            $file = _PS_DOWNLOAD_DIR_ . (string) (preg_replace('/\.{2,}/', '.', $filename));
```
Comment (Quetzacoalt91) [inline controllers/front/GetFileController.php]:
```suggestion
            $file = _PS_DOWNLOAD_DIR_ . (string) preg_replace('/\.{2,}/', '.', $filename);
```

File: controllers/front/OrderDetailController.php [hunk 4]
```diff
@@ -86,12 +86,12 @@ public function postProcess()
                     $cm->add();
 
                     if (!Configuration::get('PS_MAIL_EMAIL_MESSAGE')) {
-                        $to = strval(Configuration::get('PS_SHOP_EMAIL'));
+                        $to = (string) (Configuration::get('PS_SHOP_EMAIL'));
```
Comment (Quetzacoalt91) [inline controllers/front/OrderDetailController.php]:
```suggestion
                        $to = (string) (Configuration::get'PS_SHOP_EMAIL');
```

File: controllers/front/OrderDetailController.php [hunk 5]
```diff
@@ -86,12 +86,12 @@ public function postProcess()
...
                     } else {
                         $to = new Contact((int) Configuration::get('PS_MAIL_EMAIL_MESSAGE'));
-                        $to = strval($to->email);
+                        $to = (string) ($to->email);
```
Comment (Quetzacoalt91) [inline controllers/front/OrderDetailController.php]:
```suggestion
                        $to = (string) $to->email;
```

File: controllers/front/OrderDetailController.php [hunk 6]
```diff
@@ -86,12 +86,12 @@ public function postProcess()
...
                     }
-                    $toName = strval(Configuration::get('PS_SHOP_NAME'));
+                    $toName = (string) (Configuration::get('PS_SHOP_NAME'));
```
Comment (Quetzacoalt91) [inline controllers/front/OrderDetailController.php]:
```suggestion
                    $toName = (string) Configuration::get('PS_SHOP_NAME');
```

File: controllers/front/OrderDetailController.php [hunk 7]
```diff
@@ -120,7 +120,7 @@ public function postProcess()
                             ),
                             $to,
                             $toName,
-                            strval(Configuration::get('PS_SHOP_EMAIL')),
+                            (string) (Configuration::get('PS_SHOP_EMAIL')),
```
Comment (Quetzacoalt91) [inline controllers/front/OrderDetailController.php]:
```suggestion
                            (string) Configuration::get('PS_SHOP_EMAIL'),
```

File: src/Core/Foundation/Version.php [hunk 8]
```diff
@@ -241,8 +241,8 @@ public function isNotEqualTo($version)
     private function versionCompare($version, $operator)
     {
         if ($this->checkVersion($version)) {
-            $first = intval(trim(str_replace('.', '', $this->version)));
-            $second = intval(trim(str_replace('.', '', $version)));
+            $first = (int) (trim(str_replace('.', '', $this->version)));
```
Comment (Quetzacoalt91) [inline src/Core/Foundation/Version.php]:
```suggestion
            $first = (int) trim(str_replace('.', '', $this->version));
```

File: src/Core/Foundation/Version.php [hunk 9]
```diff
@@ -241,8 +241,8 @@ public function isNotEqualTo($version)
...
+            $second = (int) (trim(str_replace('.', '', $version)));
```
Comment (Quetzacoalt91) [inline src/Core/Foundation/Version.php]:
```suggestion
            $second = (int) trim(str_replace('.', '', $version));
```

**Comment by MathiasReker** [comment]
@Quetzacoalt91 Thank you for using suggested changes ❤️

**Comment by MathiasReker** [comment]
I don't understand why Travis fail?

**Comment by jolelievre** [comment]
seems to be the linter which blocks, maybe try to rebase (the already merged PRs) and run php-cs-fixer

**Comment by MathiasReker** [comment]
> 
> 
> seems to be the linter which blocks, maybe try to rebase (the already merged PRs) and run php-cs-fixer

hmm.. can you help me fix this? I am stock

**Comment by jolelievre** [comment]
> > seems to be the linter which blocks, maybe try to rebase (the already merged PRs) and run php-cs-fixer
> 
> hmm.. can you help me fix this? I am stock

Sure I will try this afternoon

**Comment by mickaelandrieu** [comment]
> seems to be the linter which blocks

not only the linter, but also e2e tests: sounds like you have broken something this time @MathiasReker :p

**Comment by jolelievre** [comment]
> > seems to be the linter which blocks
> 
> not only the linter, but also e2e tests: sounds like you have broken something this time @MathiasReker :p

Yes there was a syntax error in controllers/front/OrderDetailController.php, I fixed it and rebased the PR, waiting for travis right now

**Comment by MathiasReker** [comment]
@jolelievre thanks

**Comment by jolelievre** [comment]
Ok tests are good, it needs two approvals from @PrestaShop/prestashop-core-developers

**Review by mickaelandrieu — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by jolelievre** [comment]
Thank you @MathiasReker @mickaelandrieu @Quetzacoalt91

**Pull request merged**